### PR TITLE
case quick summary page without filters

### PR DESCRIPTION
**Pull request by MarkThomasLite**
### Aim

Add quick summary page as landing page when case is selected from queue view. Uses existing and new data from CaseDetail in the API.

[LTD-3671](https://uktrade.atlassian.net/browse/LTD-3671)

[LTD-3671]: https://uktrade.atlassian.net/browse/LTD-3671?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by antroy-madetech — COMMENTED** [review]
Feels like there should be more tests around the content of the Quick Summary tab. Specifics may come out once the E2E tests are working and the coverage has run. Otherwise looks good.

File: caseworker/cases/helpers/case.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
 
 
 class Tabs:
+    QUICK_SUMMARY = Tab("quick-summary", CasePage.Tabs.QUICK_SUMMARY, "quick-summary")
```
Comment (currycoder) [inline caseworker/cases/helpers/case.py:37]: I'm a little concerned that the quick summary tab appears to be the new default; I'm not sure if it's ready yet as this is the initial skeleton.  Could this also be something to check with UCD by sharing it on an environment early?

File: caseworker/templates/case/tabs/quick-summary.html [hunk 2]
```diff
@@ -0,0 +1,261 @@
+{% load static advice_tags %}
+{% load humanize %}
+
+<dl class="app-case__summary-list">
```
[the remark this answers is not in the document]
Comment (currycoder) [reply caseworker/templates/case/tabs/quick-summary.html:4]: NB, we discussed on slack subsequently that a feature flag might be a neater alternative here..

File: caseworker/cases/helpers/case.py (hunk 1 above)
Comment (MarkThomasLite) [reply caseworker/cases/helpers/case.py:37]: as discussed yest will be behind a feature flag

File: caseworker/cases/helpers/case.py [hunk 3]
```diff
@@ -127,6 +128,38 @@ def is_only_on_post_circ_queue(self):
         queue_alias = tuple(queue.get("alias") for queue in self.case.queue_details)
         return self.is_lu_user() and queue_alias == (LU_POST_CIRC_FINALISE_QUEUE_ALIAS,)
 
+    def get_goods_summary(self):
```
[the remark this answers is not in the document]
Comment (MarkThomasLite) [reply caseworker/cases/helpers/case.py:131]: yeah I tried to merge the two but with the difference in source data it just got messy

File: caseworker/cases/urls.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 app_name = "cases"
 
 urlpatterns = [
-    path("", main.CaseDetail.as_view(), name="case", kwargs={"disable_queue_lookup": True, "tab": "details"}),
```
Comment (currycoder) [inline caseworker/cases/urls.py:19]: How does this work if the feature flag is off?  Won't this still default to showing the quick summary tab by default when a caseworker clicks the case from the queue view?

File: unit_tests/caseworker/activities/test_views.py [hunk 5]
```diff
@@ -23,11 +24,20 @@ def setup(
     yield
 
 
+def test_quick_summary(authorized_client, data_queue, data_standard_case):
+    url = reverse("cases:case", kwargs={"queue_pk": data_queue["id"], "pk": data_standard_case["case"]["id"]})
+    response = authorized_client.get(url)
+    assertTemplateUsed(response, "layouts/case.html")
+    assert response.context["tabs"][0].name == "Quick summary"
+    soup = BeautifulSoup(response.content, "html.parser")
+    assert soup.find("a", {"id": "tab-quick-summary", "class": "lite-tabs__tab--selected"})
```
[the remark this answers is not in the document]
Comment (currycoder) [reply unit_tests/caseworker/activities/test_views.py:33]: What do you feel about this one Mark?

[the remark this answers is not in the document]
Comment (MarkThomasLite) [reply unit_tests/caseworker/activities/test_views.py:35]: yeah I was working on them yest, ready for review now

File: caseworker/cases/urls.py (hunk 4 above)
Comment (MarkThomasLite) [reply caseworker/cases/urls.py:19]: functionality as was when feature flag is disabled (default), feature flag was commited ready for review

File: caseworker/cases/views/main.py [hunk 6]
```diff
@@ -82,7 +81,11 @@ def get_tabs(self):
 
     def get_standard_application_tabs(self):
         tabs = self.get_tabs()
-        tabs.insert(1, Tabs.LICENCES)
+        if settings.FEATURE_QUICK_SUMMARY:
+            tabs.insert(0, Tabs.QUICK_SUMMARY)
+            tabs.insert(2, Tabs.LICENCES)
+        else:
+            tabs.insert(1, Tabs.LICENCES)
```
Comment (currycoder) [inline caseworker/cases/views/main.py:88]: Minor; Wonder if this is a leftover from an old feature flag - it should probably just be part of the list above

File: unit_tests/conftest.py [hunk 7]
```diff
@@ -1127,12 +1135,14 @@ def data_standard_case_with_potential_trigger_list_product(
...
     data["case"]["data"]["goods"][0]["is_trigger_list_guidelines_applicable"] = True
+    joined_queue_at = timezone.now() - timedelta(days=2)
     data["case"]["queue_details"].append(
         {
             "id": "566fd526-bd6d-40c1-94bd-60d10c967cf7",
             "name": "queue 20230119000000",
             "alias": "BEIS_NUCLEAR_CASES_TO_REVIEW",
-            "joined_queue_at": (timezone.now() - timedelta(days=2)).isoformat(),
+            "joined_queue_at": joined_queue_at.isoformat(),
+            "days_on_queue_elapsed": (timezone.now() - joined_queue_at).days,
```
Comment (currycoder) [inline unit_tests/conftest.py:1145]: I don't think this exists in the response anymore, right?

File: unit_tests/conftest.py [hunk 8]
```diff
@@ -515,10 +539,43 @@ def data_standard_case(
...
+                        "name": "BEIS CWC",
+                        "part_of_ecju": None,
+                        "is_ogd": False,
+                        "alias": None,
+                        "department": None,
+                    },
+                },
+            },
+            "submitted_at": submitted_at.isoformat(),
+            "total_days_elapsed": (timezone.now() - submitted_at).days,
```
Comment (currycoder) [inline unit_tests/conftest.py:578]: Is this in the API response anymore?

File: unit_tests/conftest.py [hunk 9]
```diff
@@ -515,10 +539,43 @@ def data_standard_case(
...
             "queue_names": ["queue", "queue 20200818000000"],
             "queue_details": [
-                {"id": "f458094c-1fed-4222-ac70-ff5fa20ff649", "name": "queue", "alias": "FCDO_CASES_TO_REVIEW"},
-                {"id": "c270b79b-370c-4c5e-b8b6-4d5210a58956", "name": "queue 20200818000000", "alias": "QUEUE_2"},
+                {
+                    "id": "f458094c-1fed-4222-ac70-ff5fa20ff649",
+                    "name": "queue",
+                    "alias": "FCDO_CASES_TO_REVIEW",
+                    "joined_queue_at": joined_queue_at_1.isoformat(),
+                    "days_on_queue_elapsed": (timezone.now() - joined_queue_at_1).days,
```
Comment (currycoder) [inline unit_tests/conftest.py:547]: Not in the response anymore, right?

File: unit_tests/conftest.py [hunk 10]
```diff
@@ -515,10 +539,43 @@ def data_standard_case(
...
+                },
+                {
+                    "id": "c270b79b-370c-4c5e-b8b6-4d5210a58956",
+                    "name": "queue 20200818000000",
+                    "alias": "QUEUE_2",
+                    "joined_queue_at": joined_queue_at_2.isoformat(),
+                    "days_on_queue_elapsed": (timezone.now() - joined_queue_at_2).days,
```
Comment (currycoder) [inline unit_tests/conftest.py:554]: Not in the response anymore, right?

File: unit_tests/caseworker/conftest.py [hunk 11]
```diff
@@ -334,17 +336,31 @@ def mock_standard_case(requests_mock, data_standard_case):
...
     url = client._build_absolute_uri(f"/cases/{data_standard_case['case']['id']}/")
+    joined_queue_at = timezone.now() - timedelta(days=3)
     data_standard_case["case"]["queue_details"] = [
-        {"id": "f458094c-1fed-4222-ac70-ff5fa20ff649", "name": "FCDO Countersigning", "alias": "FCDO_COUNTER_SIGNING"},
+        {
+            "id": "f458094c-1fed-4222-ac70-ff5fa20ff649",
+            "name": "FCDO Countersigning",
+            "alias": "FCDO_COUNTER_SIGNING",
+            "joined_queue_at": joined_queue_at.isoformat(),
+            "days_on_queue_elapsed": (timezone.now() - joined_queue_at).days,
```
Comment (currycoder) [inline unit_tests/caseworker/conftest.py:362]: Not in the response anymore, right?

File: unit_tests/caseworker/conftest.py [hunk 12]
```diff
@@ -334,17 +336,31 @@ def mock_standard_case(requests_mock, data_standard_case):
...
     url = client._build_absolute_uri(f"/cases/{data_standard_case['case']['id']}/")
+    joined_queue_at = timezone.now() - timedelta(days=2)
     data_standard_case["case"]["queue_details"] = [
-        {"id": "f458094c-1fed-4222-ac70-ff5fa20ff649", "name": "LU Post circulation", "alias": "LU_POST_CIRC_FINALISE"},
+        {
+            "id": "f458094c-1fed-4222-ac70-ff5fa20ff649",
+            "name": "LU Post circulation",
+            "alias": "LU_POST_CIRC_FINALISE",
+            "joined_queue_at": joined_queue_at.isoformat(),
+            "days_on_queue_elapsed": (timezone.now() - joined_queue_at).days,
```
Comment (currycoder) [inline unit_tests/caseworker/conftest.py:346]: Not in the response anymore, right?

File: caseworker/cases/views/main.py (hunk 6 above)
Comment (MarkThomasLite) [reply caseworker/cases/views/main.py:88]: ok cool I thought it was just me 😄

**Comment by MarkThomasLite** [comment]
added a comment on the one line that failed code coverage. it's temporary due to the feature flag and hard to cover so should be good to push still

**Review by saruniitr — APPROVED** [review]

**Pull request merged**